src/main/java/ca/corefacility/bioinformatics/irida/model/enums/StatisticTimePeriod.java

```diff
@@ -1,31 +1,39 @@
 package ca.corefacility.bioinformatics.irida.model.enums;
 
 /**
- * Defines a arrays of time periods used
+ * Defines a arrays of time periods and group by formats used
  * for getting statistics for the Admin Panel.
  *
  */
 
 public enum StatisticTimePeriod {
-	DAILY(7,14,30),
-	MONTHLY(90, 365),
-	YEARLY(730, 1825, 3650);
+	HOURLY(new int[] {1}, "%H:00"),
+	DAILY(new int[] {7,14,30}, "%m/%d"),
+	MONTHLY(new int[] {90, 365}, "%m/%y"),
+	YEARLY(new int[] {730, 1825, 3650}, "%Y");
 
 	private int [] values;
+	private String groupByFormat;
 
-	private StatisticTimePeriod(int... values) {
+	private StatisticTimePeriod(int[] values, String groupByFormat) {
 		this.values = values;
+		this.groupByFormat = groupByFormat;
 	}
 
-	public int[] getDaily() {
-		return DAILY.values;
+	public int[] getValues() {
+		return values;
 	}
 
-	public int[] getMonthly() {
-		return MONTHLY.values;
+	public void setValues(int[] values) {
```

Comment:
Is there any use for the setters?  Since its an enum it doesn't really make sense for an outside object to modify it.  If its needed for hibernate or something though you can switch them to `private`.

Reply:
Looks like they are not required at all. Removed in 863ac81b3bf92f01c418d0d5b77d95511d1f96df